### PR TITLE
[master] Upgrade dependencies to fix CI cron build

### LINKED ISSUE #10971: failed: nightly build failed because of binary incompatible sbt check
The nightly build is failing since we merged https://github.com/playframework/playframework/pull/10956

I didn't look further, but I suspect this is due to the upgrade to sbt 1.5.5 and the fact that the nightly build uses a akka-http snapshot. 

If I understand correct, what was previously an eviction warning is not by default an error. 

```scala
 java.lang.RuntimeException: found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error] 
[error] 	* org.scala-lang.modules:scala-java8-compat_2.13:1.0.0 (early-semver) is selected over 0.9.1
[error] 	    +- com.typesafe.play:play-streams_2.13:2.8.1+1278-9f273803-akka-2.6.16+64-bb839128-SNAPSHOT-akka-http-10.1.14+6-971cf72c-SNAPSHOT (depends on 1.0.0)
[error] 	    +- com.typesafe.akka:akka-actor_2.13:2.6.16+64-bb839128-SNAPSHOT (depends on 1.0.0)
[error] 	    +- org.scala-lang.modules:scala-java8-compat_2.13:0.9.1 (depends on 0.9.1)
[error] 
[error] 
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
[error] 	at scala.sys.package$.error(package.scala:30)
[error] 	at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:89)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:133)
[error] 	at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:73)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$20(LibraryManagement.scala:146)
[error] 	at scala.util.control.Exception$Catch.apply(Exception.scala:228)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:146)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:127)
[error] 	at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:219)
[error] 	at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:160)
[error] 	at sbt.Classpaths$.$anonfun$updateTask0$1(Defaults.scala:3678)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] 	at sbt.Execute.work(Execute.scala:291)
```

### PR DESCRIPTION
**Pull request by mkurz**
Fixes #10971

Currently cron builds fail with:
```
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
[error] (update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error] 
[error] 	* org.scala-lang.modules:scala-java8-compat_2.13:1.0.0 (early-semver) is selected over 0.9.1
[error] 	    +- com.typesafe.play:play-framework_2.13:2.8.1+1391-7eb9d1f1+20211230-1736 (depends on 1.0.0)
[error] 	    +- com.typesafe.akka:akka-actor_2.13:2.6.18           (depends on 1.0.0)
[error] 	    +- org.scala-lang.modules:scala-java8-compat_2.13:0.9.1 (depends on 0.9.1)
```

That is because of following combination:
* The `master` branch uses sbt 1.5+ already, which (in contrast to sbt <1.5) fails if it detects major version conflicts
* The cron build [uses the latest akka snaphot](https://github.com/playframework/playframework/blob/05cb03ff1d8ed37c98f1968a1bd4b4dfcf3a6250/scripts/scriptLib#L19-L30) which depends on `scala-java8-compat` 1.x, however Play depends on 0.x

Pull request builds did not fail however because they don't use the latest akka snapshot, but akka version 2.6.14, which [did not yet depend on](https://github.com/akka/akka/commit/466802c641721447826da39f29c62a7c4bcecc09) `scala-java8-compat` 1.x but still on 0.x.
Also cron builds for the 2.8.x branch did not fail because 2.8.x still is on sbt 1.3.13.

So the solution is we just need to upgrade `scala-java8-compat` to 1.0.2. And not just in Play, but also in play-ws, which luckily was done by Renato with play-ws 2.1.6 already: https://github.com/playframework/play-ws/pull/620 (That was probably because he experienced the same problem when running the scripted tests with sbt 1.5.x when he upgraded akka for the Play 2.8.11 release: https://app.travis-ci.com/github/playframework/playframework/jobs/549889172#L8473).
So basically the problem was solved for the 2.8.11 release already anyway.

I originally tried to fix this in #10969 four month ago, however back then I somehow got lost and also tried to upgrade `scala-parser-combinators` and also tried to bump play-ws, twirl and interplay to their next major versions, which actually isn't necessary. Looks like I just got confused.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ install: jabba install $(jabba ls-remote "adopt@~1.$TRAVIS_JDK.0-0" --latest=pat
 stages:
   - validations
   - test
-  - cron-test-sbt-1.5.x
+  - test-sbt-1.5.x
```
Comment (mkurz) [inline .travis.yml:32]: IMHO we should run the scripted test for each pr build, not just daily, to catched errors as early as possible. Actually that is what we did until we dropped Scala 2.12: https://github.com/playframework/playframework/pull/10956/files#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485L82
So probably I jus forgot to change that back then.

File: .travis.yml [hunk 2]
```diff
@@ -64,11 +64,17 @@ jobs:
...
+      name: "Run scripted tests (a) for sbt 1.5.x and Scala 2.13.x"
+      script: scripts/test-scripted $SCRIPTED_SBT_1_5 $TEST_SCALA_2_13 'play-sbt-plugin/*1of3'
+      workspaces:
+        use: published-local
+    - name: "Run scripted tests (b) for sbt 1.5.x and Scala 2.13.x"
+      script: scripts/test-scripted $SCRIPTED_SBT_1_5 $TEST_SCALA_2_13 'play-sbt-plugin/*2of3'
+      workspaces:
+        use: published-local
+    - name: "Run scripted tests (c) for sbt 1.5.x and Scala 2.13.x"
+      script: scripts/test-scripted $SCRIPTED_SBT_1_5 $TEST_SCALA_2_13 'play-sbt-plugin/*3of3'
```
Comment (mkurz) [inline .travis.yml:77]: Make the scripted tests run in parallel again, like we used to do: https://github.com/playframework/playframework/pull/10956/files#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485L82

File: dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/http-backend-netty-channel-options/conf/application.conf [hunk 3]
```diff
@@ -9,8 +9,8 @@ play.server {
         SO_KEEPALIVE = true
         TCP_NODELAY = true
         TCP_FASTOPEN_CONNECT = true
+        "io.netty.channel.ChannelOption#TCP_FASTOPEN" = 3
```
Comment (mkurz) [inline dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/http-backend-netty-channel-options/conf/application.conf:12]: Because the cron build for the master branch was broken since month, it never reported that this scripted test actually is broken, also see https://github.com/playframework/playframework/pull/11076#issuecomment-1002681521

File: project/Dependencies.scala [hunk 4]
```diff
@@ -80,10 +80,10 @@ object Dependencies {
   )
 
   def scalaReflect(scalaVersion: String) = "org.scala-lang" % "scala-reflect" % scalaVersion % "provided"
-  val scalaJava8Compat                   = "org.scala-lang.modules" %% "scala-java8-compat" % "0.9.1"
+  val scalaJava8Compat                   = "org.scala-lang.modules" %% "scala-java8-compat" % "1.0.2"
```
Comment (mkurz) [inline project/Dependencies.scala:83]: So this is it.

File: project/Dependencies.scala [hunk 5]
```diff
@@ -263,7 +263,7 @@ object Dependencies {
     "com.github.ben-manes.caffeine" % "jcache"   % caffeineVersion
   ) ++ jcacheApi
 
-  val playWsStandaloneVersion = "2.1.3"
+  val playWsStandaloneVersion = "2.1.6"
```
Comment (mkurz) [inline project/Dependencies.scala:266]: Here was well.

**Comment by mkurz** [comment]
Blocked by https://github.com/akka/akka/pull/30871 unfortunately.

**Comment by mkurz** [comment]
Will hopefully be unblocked by https://github.com/akka/akka/pull/31046

File: .travis.yml [hunk 6]
```diff
@@ -29,7 +29,7 @@ install: jabba install $(jabba ls-remote "adopt@~1.$TRAVIS_JDK.0-0" --latest=pat
 stages:
   - validations
   - test
-  - cron-test-sbt-1.5.x
+  - test-sbt-1.6.x
```
Comment (mkurz) [inline .travis.yml:32]: IMHO we should run the scripted test for each pr build, not just daily, to catched errors as early as possible. Actually that is what we did until we dropped Scala 2.12: https://github.com/playframework/playframework/pull/10956/files#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485L82
So probably I just forgot to change that back then.

File: .travis.yml [hunk 7]
```diff
@@ -64,11 +64,17 @@ jobs:
...
+      name: "Run scripted tests (a) for sbt 1.6.x and Scala 2.13.x"
+      script: scripts/test-scripted $SCRIPTED_SBT_1_6 $TEST_SCALA_2_13 'play-sbt-plugin/*1of3'
+      workspaces:
+        use: published-local
+    - name: "Run scripted tests (b) for sbt 1.6.x and Scala 2.13.x"
+      script: scripts/test-scripted $SCRIPTED_SBT_1_6 $TEST_SCALA_2_13 'play-sbt-plugin/*2of3'
+      workspaces:
+        use: published-local
+    - name: "Run scripted tests (c) for sbt 1.6.x and Scala 2.13.x"
+      script: scripts/test-scripted $SCRIPTED_SBT_1_6 $TEST_SCALA_2_13 'play-sbt-plugin/*3of3'
```
Comment (mkurz) [inline .travis.yml:77]: Make the scripted tests run in parallel again, like we used to do: https://github.com/playframework/playframework/pull/10956/files#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485L82

File: project/Dependencies.scala [hunk 8]
```diff
@@ -7,16 +7,16 @@ import Keys._
 import buildinfo.BuildInfo
 
 object Dependencies {
-  val akkaVersion: String = sys.props.getOrElse("akka.version", "2.6.14")
-  val akkaHttpVersion     = sys.props.getOrElse("akka.http.version", "10.1.14")
+  val akkaVersion: String = sys.props.getOrElse("akka.version", "2.6.18+21-4fb7bd9b-SNAPSHOT")
```
Comment (mkurz) [inline project/Dependencies.scala:10]: Only temporary, let's use this akka snapshot because it contains https://github.com/akka/akka/pull/31046 to unblock this PR.

File: transport/client/play-ahc-ws/src/main/java/play/libs/ws/ahc/AhcWSRequest.java [hunk 9]
```diff
@@ -332,6 +332,11 @@ public WSRequest setFollowRedirects(boolean followRedirects) {
     return converter.apply(request.setFollowRedirects(followRedirects));
   }
 
+  @Override
+  public WSRequest setDisableUrlEncoding(boolean disableUrlEncoding) {
+    return converter.apply(request.setDisableUrlEncoding(disableUrlEncoding));
+  }
+
```
Comment (mkurz) [inline transport/client/play-ahc-ws/src/main/java/play/libs/ws/ahc/AhcWSRequest.java:339]: These methods were added as part of https://github.com/playframework/play-ws/pull/552

File: core/play-integration-test/src/it/scala/play/it/http/UriHandlingSpec.scala [hunk 10]
```diff
@@ -52,14 +52,15 @@ class UriHandlingSpec
...
-
+    /*
     "handle '/pat/resources/BodhiApplication?where={%22name%22:%22hsdashboard%22}' as a valid URI" in makeRequest(
       "/pat/resources/BodhiApplication?where={%22name%22:%22hsdashboard%22}"
     ) {
       case (endpoint, response) => {
         response.body.string must_=== """/pat/resources/BodhiApplication?where={"name":"hsdashboard"}"""
       }
     }
+     */
```
Comment (mkurz) [inline core/play-integration-test/src/it/scala/play/it/http/UriHandlingSpec.scala:63]: For now I disable theses it tests, because they are breaking after upgrading to akka-http 10.2.x.
To move on lets fix that later, opened #11113

File: dev-mode/sbt-scripted-tools/src/main/scala/play/sbt/scriptedtools/ScriptedTools.scala [hunk 11]
```diff
@@ -30,6 +30,7 @@ object ScriptedTools extends AutoPlugin {
 
   override def projectSettings: Seq[Def.Setting[_]] = Def.settings(
     resolvers += Resolver.sonatypeRepo("releases"), // sync BuildSettings.scala
+    resolvers += Resolver.sonatypeRepo("snapshots"),
```
Comment (mkurz) [inline dev-mode/sbt-scripted-tools/src/main/scala/play/sbt/scriptedtools/ScriptedTools.scala:33]: We can remove that one later, see #11114

File: dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/evolutions-auto-apply-false/conf/evolutions/default/1.sql [hunk 12]
```diff
@@ -3,7 +3,7 @@
 # --- !Ups
 
 CREATE TABLE users (
-    id bigint(20) NOT NULL AUTO_INCREMENT,
+    id bigint GENERATED BY DEFAULT AS IDENTITY,
```
Comment (mkurz) [inline dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/evolutions-auto-apply-false/conf/evolutions/default/1.sql:6]: Necessary because master branch upgraded h2 from 1.x to 2.x, see https://stackoverflow.com/questions/70695039/org-h2-jdbc-jdbcsqlsyntaxerrorexception-after-h2-version-upgrade/70696184

**Review by octonato — APPROVED** [review]
LGTM
